lib/matplotlib/figure.py

```diff
@@ -2151,22 +2153,67 @@ def __init__(self,
             parameters :rc:`figure.subplot.*` are used.
 
         tight_layout : bool or dict, default: :rc:`figure.autolayout`
-            If ``False`` use *subplotpars*. If ``True`` adjust subplot
-            parameters using `.tight_layout` with default padding.
-            When providing a dict containing the keys ``pad``, ``w_pad``,
-            ``h_pad``, and ``rect``, the default `.tight_layout` paddings
-            will be overridden.
+            This is equal to ``layout='tight'`` or ``layout=dict(...)``.
+
+            .. admonition:: Discouraged
+
+                The use of this parameter is discouraged. Please use
+                ``layout='tight'`` or ``layout=dict(...)`` instead.
 
         constrained_layout : bool, default: :rc:`figure.constrained_layout.use`
-            If ``True`` use constrained layout to adjust positioning of plot
-            elements.  Like ``tight_layout``, but designed to be more
-            flexible.  See
-            :doc:`/tutorials/intermediate/constrainedlayout_guide`
-            for examples.  (Note: does not work with `add_subplot` or
-            `~.pyplot.subplot2grid`.)
+            This is equal to ``layout='constrained'``.
+
+            .. admonition:: Discouraged
+
+                The use of this parameter is discouraged. Please use
+                ``layout='constrained'`` instead.
+
+        layout : {'constrained', 'tight'} or dict, optional
```

Comment:
Unfortunately, constrained_layout also accepts a dict as documented in `set_constrained_layout`....

Reply:
Hm, have to think of a workaround for that.

Is that something really used? I haven't seen any example of it in the docs.
If it's only for some very rare cases we could skip that functionality in the figure creation and only support it in `set_constrained_layout()`.

Reply:
It is exactly analogous to what is done for `tight_layout`, so I don't see why we would let one have a dict, but not the other.  I'm fine if *neither* has a dict.  I don't use the dict, but then I wrote constrained_layout, so the spacings are OK for me :wink:  I agree that it is not well documented, even in the guide.

Reply:
Seems, that *neither* is the way to go then.